internal/docs/exported_fields.go

```diff
@@ -23,15 +23,17 @@ type fieldsTableRecord struct {
 var escaper = strings.NewReplacer("*", "\\*", "{", "\\{", "}", "\\}", "<", "\\<", ">", "\\>")
 
 func renderExportedFields(fieldsParentDir string) (string, error) {
-	validator, err := fields.CreateValidatorForDirectory(fieldsParentDir)
+	injectOptions := fields.InjectFieldsOptions{
+		// Keep External parameter when rendering fields, so we can render
+		// documentation for empty groups imported from ECS, for backwards compatibility.
+		KeepExternal: true,
```

Comment:
This parameter for fields injection can be a bit difficult to follow. Its only purpose is to mimic a collateral behaviour we had with lazy resolution of external fields: In some places we were skipping group fields when they were empty. But this was not happening with external fields, because on these checks the field was still unresolved.

We may argue that this behaviour is unexpected and we should fix it instead, because it treats fields on different ways depending if they are external or not. But fixing this would break CI builds of packages, because it changes what fields are rendered.

This flag helps keeping the old behaviour for rendered documentation.:
* When rendering documentation, we keep external, so we can keep the old behaviour.
* In other cases, such as when resolving external fields in package builds, we remove the "external" parameter once the field is injected, so it is not written to fields files. On these cases we may be removing empty groups from built fields files, but this is actually good because they don't generate any meaningful mapping.